cli/azd/pkg/config/manager.go

```diff
@@ -109,7 +110,28 @@ func GetUserConfigDir() (string, error) {
 		configDirPath = filepath.Join(homeDir, configDir)
 	}
 
-	err := os.MkdirAll(configDirPath, osutil.PermissionDirectory)
+	err := os.MkdirAll(configDirPath, osutil.PermissionDirectoryOwnerOnly)
```

Comment:
Haven't really thought through the permission change here... We've been pretty relaxed in stamping directories with `755` in the past, `700` seems right here, but I'm not sure what may break... @ellismg @vhvb1989 @wbreza ?

Reply:
I think 700 is a reasonable default here for the current user. We are more restrictive in our file permissions in other places, but those are where sensitive information is being cached: https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/auth/manager.go#L86 


I would expect that we'd want the same permissions in `~/.azd` to begin with.

Reply:
I think either choice is reasonable (looking at my mac for example, it seems like many tools will use 0755).  I explicitly denied group and world access to the auth related as a best practice and defense in depth thing.

In practice, I think it would be fine to not grant group and world access to anything we create under `~` (I think it's safe to assume that if HOME resolves to something it resolves to a thing that the current user has access to, and so we don't need to worry about group and world permissions being the thing that grants us access).

Reply:
Defense in depth issue opened -- https://github.com/Azure/azure-dev/issues/1991